OneSignalSDK/onesignal/src/main/java/com/onesignal/OSNotificationReceivedEvent.java

```diff
@@ -77,6 +77,21 @@ public synchronized void complete(@Nullable OSNotification notification) {
 
       isComplete = true;
 
+      // Logic for Wrappers
```

Comment:
Logic isn't specifically for wrappers. A native developer could call the complete from the main thread as well.

Reply:
yeah, but will need to specially make it from the main thread, on native the service extender is being called from a worker